src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -2442,18 +2446,195 @@ private byte[] AsExactByteCount(int size)
+                    if (_targetFields[i] is not FieldDesc targetField)
+                    {
+                        builder.EmitZeroPointer();
+                    }
+                    else
+                    {
+                        Debug.Assert(targetField.IsStatic && !targetField.HasGCStaticBase && !targetField.IsThreadStatic && !targetField.HasRva);
+                        ISymbolNode nonGcStaticBase = factory.TypeNonGCStaticsSymbol((MetadataType)targetField.OwningType);
+                        builder.EmitPointerReloc(nonGcStaticBase, targetField.Offset.AsInt);
+                    }
+                }
+            }
+
+            public override bool GetRawData(NodeFactory factory, out object data)
+            {
+                data = null;
+                return false;
+            }
+
+            public override bool TryCreateByRef(out Value value)
+            {
+                value = new ComInterfaceEntrySlotReference(_entryType, _targetFields, _guidBytes, 0);
+                return true;
+            }
+
+            private sealed class ComInterfaceEntrySlotReference : ByRefValueBase, IHasInstanceFields
+            {
+                private readonly MetadataType _entryType;
+                private readonly FieldDesc[] _targetFields;
+                private readonly byte[][] _guidBytes;
+                private readonly int _index;
+
+                public ComInterfaceEntrySlotReference(MetadataType entryType, FieldDesc[] targetFields, byte[][] guidBytes, int index)
+                    => (_entryType, _targetFields, _guidBytes, _index) = (entryType, targetFields, guidBytes, index);
+
+                public override bool TryCompareEquality(Value value, out bool result)
+                {
+                    result = false;
+                    return false;
+                }
+
+                public override bool GetRawData(NodeFactory factory, out object data)
+                {
+                    data = null;
+                    return false;
+                }
+
+                public override void WriteFieldData(ref ObjectDataBuilder builder, NodeFactory factory)
+                {
+                    throw new NotSupportedException();
+                }
+
+                bool IHasInstanceFields.TrySetField(FieldDesc field, Value value)
+                {
+                    if (field.OwningType != _entryType)
+                        return false;
+
+                    if (field.Name == "IID"
+                        && value is ValueTypeValue guidValue
+                        && guidValue.Size == _guidBytes[_index].Length)
+                    {
+                        Array.Copy(guidValue.InstanceBytes, _guidBytes[_index], _guidBytes[_index].Length);
+                        return true;
+                    }
+                    else if (field.Name == "Vtable"
+                        && value is ByRefValueBase byrefValue
+                        && byrefValue.MemoryOwner != null)
```

Comment:
Would `BackingField` (or something else that has some association with a field) be a better name for `MemoryOwner`. I have to keep reminding myself when reading this that MemoryOwner is actually a field where the item persisted.